packages/pluggableWidgets/datagrid-date-filter-web/src/utils/utils.ts

```diff
@@ -13,3 +13,37 @@ export function changeTimeToMidnight(date: Date): Date {
     newDate.setMilliseconds(0);
     return newDate;
 }
+
+/**
+ * This function takes format string and replace
+ * single "d" and "M" with their "double" equivalent
+ * Main purpose is to "fix" value formats accepted
+ * by DateFilter input. Idea is that if format string is
+ * uses short format (e.g d-M-yyyy) user should still
+ * be able to type value with leading zeros "09-02-2002".
+ *
+ * Example:
+ * "d/M/yyyy" -> "dd/MM/yyyy"
+ *
+ * @param formatString
+ */
+export function doubleMonthOrDayWhenSingle(formatString: string): string {
+    return Array.from(formatString)
+        .reduce<string[]>((acc, char, index, array) => {
+            if (char === "d" || char === "M") {
+                const isFirst = index === 0;
+                const isLast = index === array.length - 1;
+                const isNotEqualPrev = isFirst || char !== array[index - 1];
+                const isNotEqualNext = isLast || char !== array[index + 1];
+
+                if (isNotEqualPrev && isNotEqualNext) {
+                    acc.push(char + char);
+                    return acc;
+                }
+            }
+
+            acc.push(char);
+            return acc;
+        }, [])
+        .join("");
+}
```

Comment:
Is there a reason why you cannot use `replaceAll` with a replacer function? This looks very low-level and complicated.

Reply:
> Is there a reason why you cannot use `replaceAll` with a replacer function? This looks very low-level and complicated.

But if we going to use `replaceAll` then we have to create regex that will find all matches that we need to replace. After few attempts I was not able to find good regex alternative that will work in Safari and decided to switch to simple function. If you think we should go with `replaceAll` I would ask for help on finding regex that works. @tamelbrun

Reply:
Wow 🤯 , yeah, that's much simpler. I didn't get it from first time.

Reply:
All test passed 😮 😎 .

Reply:
👍🏻✨